### ISSUE TITLE
False-positive 'unused-import' for base class name used in type annotation

### ISSUE THREAD

**Issue by saf** [PyCQA/pylint#3191]
Starting from pylint 2.4.0, an imported symbol from my own package which is only used in a `# type:` annotation has been unexpectedly reported as an `unused-import`. 

This works fine with pylint 2.3.1.

### Steps to reproduce
`mod1/my_types.py`:
```
class Base():
    pass

class Foo(Base):
    pass

class Bar(Base):
    pass
```
`mod1/usage.py`:
```
#!/usr/bin/python3

import sys

from mod1.my_types import Base, Foo, Bar

def main():
    if sys.argv[1:]:
        foo_or_bar = Foo()  # type: Base
    else:
        foo_or_bar = Bar()

    print(foo_or_bar.__class__)


if __name__ == "__main__":
    main()
```
### Current behavior
```
mod1/usage.py:5: [W0611: unused-import] Unused Base imported from mod1.my_types
```

### Expected behavior
No error reported about the unused import.

### pylint --version output
```
$ python3 -m pylint --version
pylint 2.4.2
astroid 2.3.1
Python 3.6.8 (default, Oct  7 2019, 12:59:55) 
[GCC 8.3.0]
```

**Comment by PCManticore** [comment]
Thanks, I can reproduce the issue.
